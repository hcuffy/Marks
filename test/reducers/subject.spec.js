import applyClassList, { applySubjectData } from '../../app/reducers/subjectReducer'
import {
	UPDATE_CLASS_LIST,
	GET_SINGLE_SUBJECT,
	GET_SUBJECT_LIST,
	ADD_NEW_SUBJECT
} from '../../app/constants/actionTypes'

const initialLoadState = {
	openModal: false,
	subject: null,
	name: 'name',
	abbreviation: 'abbreviation'
}

describe('test subject reducer', () => {
	it('should use default class state', () => {
		expect(applyClassList(initialLoadState, {})).toMatchSnapshot()
	})

	it('should update subject list', () => {
		const payload = 'Science'
		expect(
			applyClassList(initialLoadState, { type: UPDATE_CLASS_LIST, payload })
		).toMatchSnapshot()
	})

	it('should get sinlge subject', () => {
		const payload = 'Mathematics'
		expect(
			applyClassList(initialLoadState, { type: GET_SINGLE_SUBJECT, payload })
		).toMatchSnapshot()
	})

	it('should use default subject state', () => {
		expect(applyClassList(initialLoadState, {})).toMatchSnapshot()
	})

	it('should get subject list', () => {
		const data = [
			{
				name: 'Biology',
				abbreviation: 'BIO101',
				room: 'Room 2',
				tests: [],
				classroomId: 'qPtCHoVmf2Tbznji',
				_id: 'qGK99ne2xYjkW4cp',
				createdAt: { $$date: 1534404077665 },
				updatedAt: { $$date: 15544404096125 }
			}, {
				name: 'English',
				abbreviation: 'ENG101',
				room: 'Room 2',
				tests: [],
				classroomId: 'qPtCHnohVmf2Tbznji',
				_id: 'upGQBZVHHM4J1Tam',
				createdAt: { $$date: 1555383021646 },
				updatedAt: { $$date: 1553871237090 }
			}
		]
		const payload = { data }
		expect(
			applySubjectData(initialLoadState, { type: GET_SUBJECT_LIST, payload })
		).toMatchSnapshot()
	})

	it('should add new subject', () => {
		const data = [
			{
				name: 'Mathematics',
				abbreviation: 'MATH101',
				room: 'Room 2',
				tests: [],
				classroomId: 'qPtCHoVmf2Tbznji',
				_id: 'lpfC4fTZERCjwfNo',
				createdAt: { $$date: 1556465049016 },
				updatedAt: { $$date: 1556465049016 }
			}
		]
		const payload = { data }
		expect(
			applySubjectData(initialLoadState, { type: ADD_NEW_SUBJECT, payload })
		).toMatchSnapshot()
	})
})
