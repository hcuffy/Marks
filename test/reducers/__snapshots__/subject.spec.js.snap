// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test subject reducer should add new subject 1`] = `
Object {
  "abbreviation": "abbreviation",
  "data": Array [
    Object {
      "_id": "lpfC4fTZERCjwfNo",
      "abbreviation": "MATH101",
      "classroomId": "qPtCHoVmf2Tbznji",
      "createdAt": Object {
        "$$date": 1556465049016,
      },
      "name": "Mathematics",
      "room": "Room 2",
      "tests": Array [],
      "updatedAt": Object {
        "$$date": 1556465049016,
      },
    },
  ],
  "name": "name",
  "openModal": false,
  "subject": null,
}
`;

exports[`test subject reducer should get sinlge subject 1`] = `
Object {
  "abbreviation": "abbreviation",
  "name": "name",
  "openModal": false,
  "subject": "Mathematics",
}
`;

exports[`test subject reducer should get subject list 1`] = `
Object {
  "abbreviation": "abbreviation",
  "data": Array [
    Object {
      "_id": "qGK99ne2xYjkW4cp",
      "abbreviation": "BIO101",
      "classroomId": "qPtCHoVmf2Tbznji",
      "createdAt": Object {
        "$$date": 1534404077665,
      },
      "name": "Biology",
      "room": "Room 2",
      "tests": Array [],
      "updatedAt": Object {
        "$$date": 15544404096125,
      },
    },
    Object {
      "_id": "upGQBZVHHM4J1Tam",
      "abbreviation": "ENG101",
      "classroomId": "qPtCHnohVmf2Tbznji",
      "createdAt": Object {
        "$$date": 1555383021646,
      },
      "name": "English",
      "room": "Room 2",
      "tests": Array [],
      "updatedAt": Object {
        "$$date": 1553871237090,
      },
    },
  ],
  "name": "name",
  "openModal": false,
  "subject": null,
}
`;

exports[`test subject reducer should update subject list 1`] = `
Object {
  "abbreviation": "abbreviation",
  "name": "name",
  "openModal": true,
  "subject": "Science",
}
`;

exports[`test subject reducer should use default class state 1`] = `
Object {
  "abbreviation": "abbreviation",
  "name": "name",
  "openModal": false,
  "subject": null,
}
`;

exports[`test subject reducer should use default subject state 1`] = `
Object {
  "abbreviation": "abbreviation",
  "name": "name",
  "openModal": false,
  "subject": null,
}
`;
